linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -2793,6 +2860,46 @@ private class FunctionEmitter private (
     AnyType
   }
 
+  private def genRecordSelect(tree: RecordSelect): Type = {
+    if (canLookupRecordSelect(tree)) {
+      markPosition(tree)
+      genReadStorage(lookupRecordSelect(tree))
+    } else {
```

Comment:
It is unclear to me what the else branch is for. It looks like it is handling structural subtyping for records.

However, if this is the case:
- Why does this only need to happen when we select on a record (and not whenever we pass a record "upwards").
- Why does `Type#isSubtype` not express this relationship?

Reply:
This is not about subtyping. I added a comment explaining the situation and how it differs from the JS backend.